components/constellation/constellation.rs

```diff
@@ -2904,15 +2906,29 @@ where
             self.pressed_mouse_buttons = 0;
         }
 
-        let msg = ConstellationControlMsg::SendEvent(destination_pipeline_id, event);
-        let result = match self.pipelines.get(&destination_pipeline_id) {
+        let pipeline = match self.pipelines.get(&destination_pipeline_id) {
             None => {
                 debug!("{}: Got event after closure", destination_pipeline_id);
                 return;
             },
-            Some(pipeline) => pipeline.event_loop.send(msg),
+            Some(pipeline) => pipeline,
         };
-        if let Err(e) = result {
+        let event_variant = match event {
+            CompositorEvent::ResizeEvent(..) => CompositorEventVariant::ResizeEvent,
+            MouseButtonEvent(..) => CompositorEventVariant::MouseButtonEvent,
+            MouseMoveEvent(..) => CompositorEventVariant::MouseMoveEvent,
+            CompositorEvent::TouchEvent(..) => CompositorEventVariant::TouchEvent,
+            CompositorEvent::WheelEvent(..) => CompositorEventVariant::WheelEvent,
+            CompositorEvent::KeyboardEvent(..) => CompositorEventVariant::KeyboardEvent,
+            CompositorEvent::CompositionEvent(..) => CompositorEventVariant::CompositionEvent,
+            CompositorEvent::IMEDismissedEvent => CompositorEventVariant::IMEDismissedEvent,
+        };
+        let msg = EmbedderMsg::EventDelivered(event_variant);
+        self.embedder_proxy
+            .send((Some(pipeline.top_level_browsing_context_id), msg));
```

Comment:
I would avoid the temporary `msg` here.

Reply:
done in 5f9b48d37f82f63e9da7feca448a26b1493028dc

Reply:
done in 5f9b48d37f82f63e9da7feca448a26b1493028dc